src/ucs/arch/x86_64/cpu.c

```diff
@@ -707,6 +736,319 @@ ucs_status_t ucs_arch_get_cache_size(size_t *cache_sizes)
+                y3 = _mm256_loadu_si256(UCS_PTR_BYTE_OFFSET(src, offset + 96));
+                _mm256_stream_si256(UCS_PTR_BYTE_OFFSET(dst, offset), y0);
+                _mm256_stream_si256(UCS_PTR_BYTE_OFFSET(dst, offset + 32), y1);
+                _mm256_stream_si256(UCS_PTR_BYTE_OFFSET(dst, offset + 64), y2);
+                _mm256_stream_si256(UCS_PTR_BYTE_OFFSET(dst, offset + 96), y3);
+
+                if (ucs_unlikely(hint & UCS_ARCH_MEMCPY_NT_SOURCE)) {
+                    ucs_nt_read_prefetch(UCS_PTR_BYTE_OFFSET(src, offset + 3 * 64));
+                    ucs_nt_read_prefetch(UCS_PTR_BYTE_OFFSET(src, offset + 4 * 64));
+                } else {
+                    if (len > 256) {
+                        ucs_read_prefetch(UCS_PTR_BYTE_OFFSET(src, offset + 3 * 64));
+                        ucs_read_prefetch(UCS_PTR_BYTE_OFFSET(src, offset + 4 * 64));
+                    }
+                }
+
+                offset += 128;
+                len    -= 128;
+            }
+        } else {
+            /* src address aligned to 32 byte */
+            while (len >= 128) {
+                y0 = _mm256_load_si256(UCS_PTR_BYTE_OFFSET(src, offset));
+                y1 = _mm256_load_si256(UCS_PTR_BYTE_OFFSET(src, offset + 32));
+                y2 = _mm256_load_si256(UCS_PTR_BYTE_OFFSET(src, offset + 64));
+                y3 = _mm256_load_si256(UCS_PTR_BYTE_OFFSET(src, offset + 96));
+                _mm256_stream_si256(UCS_PTR_BYTE_OFFSET(dst, offset), y0);
+                _mm256_stream_si256(UCS_PTR_BYTE_OFFSET(dst, offset + 32), y1);
+                _mm256_stream_si256(UCS_PTR_BYTE_OFFSET(dst, offset + 64), y2);
+                _mm256_stream_si256(UCS_PTR_BYTE_OFFSET(dst, offset + 96), y3);
+
+                if (ucs_unlikely(hint & UCS_ARCH_MEMCPY_NT_SOURCE)) {
+                    ucs_nt_read_prefetch(UCS_PTR_BYTE_OFFSET(src, offset + 3 * 64));
+                    ucs_nt_read_prefetch(UCS_PTR_BYTE_OFFSET(src, offset + 4 * 64));
+                } else {
+                    if (len > 256) {
+                        ucs_read_prefetch(UCS_PTR_BYTE_OFFSET(src, offset + 3 * 64));
+                        ucs_read_prefetch(UCS_PTR_BYTE_OFFSET(src, offset + 4 * 64));
+                    }
+                }
+
+                offset += 128;
+                len    -= 128;
+            }
+        }
+
+        if (len >= 64) {
+            y0 = _mm256_loadu_si256(UCS_PTR_BYTE_OFFSET(src, offset));
+            y1 = _mm256_loadu_si256(UCS_PTR_BYTE_OFFSET(src, offset + 32));
+            _mm256_stream_si256(UCS_PTR_BYTE_OFFSET(dst, offset), y0);
+            _mm256_stream_si256(UCS_PTR_BYTE_OFFSET(dst, offset + 32), y1);
+            offset += 64;
+            len    -= 64;
+        }
+
+        if (len) {
+            ucs_nt_write_prefetch(UCS_PTR_BYTE_OFFSET(dst, offset));
+        }
+
+        /* make the writes visible to the other core */
+        _mm_sfence();
```

Comment:
is there a UCS macro for this?

Reply:
Yea there is one "ucs_memory_bus_store_fence()" I missed it somehow :)

Reply:
Done.